src/features/messages/cards/TransactionCard.tsx

```diff
@@ -207,10 +208,15 @@ function DeliveryStatus({ children }: PropsWithChildren<unknown>) {
   );
 }
 
-function CallDataModal({ debugResult }: { debugResult?: MessageDebugResult }) {
+function CallDataModal({ debugResult,chainId }: { debugResult?: MessageDebugResult,chainId:ChainId }) {
   const [isOpen, setIsOpen] = useState(false);
+  const disabled=false
+  const [buttonText,setButtonText]=useState<string>("Simulate Handle Call")
   if (!debugResult?.calldataDetails) return null;
   const { contract, handleCalldata } = debugResult.calldataDetails;
+  function handleClick() {
+    setButtonText('Simulating');
```

Comment:
Please show a spinner while it's simulating in addition to changing button text.